### PR TITLE
websocket stress example

### PR DESCRIPTION
**Pull request by neonphog**
Adds a long-running example executable that starts a holochain conductor and spawns both longer and shorter running websocket connections that both close gracefully and make a request and drop immediately.

I see no errors, and no performance problems when run through linux perf. Leading me to believe the problems we were seeing with the dweb hApp were either an error edge-case not reproduced by this stress test, a problem downstream of holochain, or (I think most likely) were *actually* just a visible symptom of the overload holochain was experiencing during the connection / request flooding.

### CODE REVIEW

File: crates/holochain/examples/websocket_stress.rs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+static CONS_MADE: AtomicU64 = AtomicU64::new(0);
+static MSGS: AtomicU64 = AtomicU64::new(0);
+static GOOD_CLOSE: AtomicU64 = AtomicU64::new(0);
+static BAD_CLOSE: AtomicU64 = AtomicU64::new(0);
+
+#[tokio::main(flavor = "multi_thread")]
+pub async fn main() {
+    let tmp_dir = TempDir::new().unwrap();
+    let environment_path = tmp_dir.path().to_path_buf();
+    let config = ConductorConfig {
```
Comment (steveej) [inline crates/holochain/examples/websocket_stress.rs:38]: @thedavidmeister where does a `ConductorConfig` fit in with your network toplogy model?

File: crates/holochain/examples/websocket_stress.rs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+const MINUTES_LONG_BEHAVED_COUNT: usize = 10;
+const MINUTES_LONG_BAD_COUNT: usize = 10;
+const SECONDS_LONG_BEHAVED_COUNT: usize = 100;
+const SECONDS_LONG_BAD_COUNT: usize = 100;
+
+static CONS_MADE: AtomicU64 = AtomicU64::new(0);
+static MSGS: AtomicU64 = AtomicU64::new(0);
+static GOOD_CLOSE: AtomicU64 = AtomicU64::new(0);
+static BAD_CLOSE: AtomicU64 = AtomicU64::new(0);
```
[the remark this answers is not in the document]
Comment (ThetaSinner) [reply crates/holochain/examples/websocket_stress.rs:32]: This does look like metrics to me. I would like to see these embedded in Holochain under the `otel` feature flag rather than measured from here. I've deliberately held off re-targeting the metrics work from `develop-0.1` to `develop` and preparing for merge because I don't want to be putting anything big/risky into `develop` like new dependencies and feature flags while we're trying to focus on bug fixes but would that be useful to anybody if we put some time into it?

**Comment by ThetaSinner** [comment]
There have been some other efforts on websocket testing
- A heavy usage test from me https://github.com/holochain/holochain/pull/2452
- An as-yet-unmerged stress test from @maackle https://github.com/holochain/holochain/pull/2514

It would be good to unify approaches a bit - so could this be a test and is there an appropriate place/time that this could be run along with my test that is disabled because it's unreasonably slow for CI?
But also, can we all collect metrics in a similar way? Because it's hard to interpret what the tests are telling us without some uniformity of output. The analysis can be as unique as we need, but consistent ways to run and observe feels important.

Not necessarily all problems for this PR, but keen to raise this

File: crates/holochain/examples/websocket_stress.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (neonphog) [reply crates/holochain/examples/websocket_stress.rs:32]: I'm certainly up for updating these into some standardized form once we've come up with that standardization. But, in terms of bang-for-buck, this here is an experiment that currently must be manually run to understand the results. I don't think reporting the metrics anywhere provides any benefit.

**Comment by neonphog** [comment]
@ThetaSinner @maackle @steveej 

@abe-njama had asked me to look into the dweb websocket problems per [this story](https://www51.v1host.com/Holo/defect.mvc/Summary?oidToken=Defect%3A71430). I hadn't realized there were other efforts. I'm totally fine not merging this if there's no need for it. For me it already served it's purpose in deciding it's more about the overload than the websockets anyways.

> It would be good to unify approaches a bit - so could this be a test and is there an appropriate place/time that this could be run along with my test that is disabled because it's unreasonably slow for CI? But also, can we all collect metrics in a similar way?

Indeed, we have not yet invested in any type of framework for long-running tests, and thus anything long-running must be manually observed. I don't see much benefit to unifying anything unless we're ready to invest in that framework which makes the unification useful. So, I'm up for having that conversation if that's where we're at : )

**Review by jost-s — APPROVED** [review]

**Comment by ThetaSinner** [comment]
I think this does cover something the other tests don't. I'm just wondering if it can be a test rather than an example. Maybe I'm wrong but I'd expect us to use the default test runner for Rust (or one compatible with the same annotations anyway) to run 'very slow' or otherwise long running tests. The difference would be that they'd need a real environment to talk to and have a 'harness' of some kind to allow interactions like Sweettest provides but against an environment.

I am just thinking this could be closer to that end goal by being an ignored test but if that doesn't sound like a good idea then this is fine as it is

**Review by ThetaSinner — APPROVED** [review]

**Comment by neonphog** [comment]
> I'm just wondering if it can be a test rather than an example

The original idea was for it to run indefinitely, which obviously isn't compatible with it being a test. But I suppose there's no harm in applying an arbitrary timeout that can always be change in the moment we need it to run longer. Will do.
